server/src/server.js

```diff
@@ -0,0 +1,166 @@
+const { v4: uuidv4 } = require("uuid");
+const express = require("express");
+const cookieSession = require("cookie-session");
+const datastore = require("./datastore");
+const app = express();
+
+const KIND_EVENT = "Event";
+
+const URL_PARAM_EVENT_ID = `eventID`;
+
+const ERROR_BAD_DB_INTERACTION = "BAD_DATABASE";
+const ERROR_INVALID_EVENT_ID = "INVALID_EVENT_ID";
+const ERROR_BAD_UUID = "BAD_UUID";
+
+// Parse request bodies with the json content header into JSON
+app.use(express.json());
+
+app.use(
+  cookieSession({
+    name: "session",
+    keys: ["secret1"],
+    maxAge: 365 * 24 * 60 * 60 * 1000, // 1 year
+  })
+);
+
+/**
+ * Middleware that attaches to every request and checks if there
+ * is a user id. If not, we assume this is a new user and give
+ * them a new uuid that is stored in their cookie.
+ */
+app.use((request, response, next) => {
+  if (request.session.isNew) {
+    // Don't worry about collision
+    const uuid = uuidv4();
+    request.session.userID = uuid;
+  } else {
+    if (
+      // regex that matches valid uuids
+      !/^[a-z0-9]{8}-[a-z0-9]{4}-4[a-z0-9]{3}-[a-z0-9]{4}-[a-z0-9]{12}$/.test(
```

Comment:
What is the purpose of this?

Reply:
Each user is assigned a unique UUID which is stored in a cookie. This regex ensures any existing UUID in a cookie is valid.